libdieharder/rng_file_input.c

```diff
@@ -325,16 +325,22 @@ static void file_input_set (void *vstate, unsigned long int s)
        }
      }
      if(strncmp(splitbuf[0],"count",5) == 0){
-       state->flen = atoi(splitbuf[1]);
+       if(sscanf(splitbuf[1], "%"SCNd64, &state->flen) != 1){
+         fprintf(stderr,"# file_input(): Error: invalid count value\n");
+         exit(0);
+       }
        filecount = state->flen;
        cnt++;
-       if(verbose){ 
+       if(verbose){
          fprintf(stdout,"# file_input(): cnt = %d\n",cnt);
-         fprintf(stdout,"# file_input(): state->flen set to %lu\n",state->flen);
+         fprintf(stdout,"# file_input(): state->flen set to %"PRId64"\n",state->flen);
        }
      }
      if(strncmp(splitbuf[0],"numbit",6) == 0){
-       filenumbits = atoi(splitbuf[1]);
```

Comment:
one more comment - filenumbits is int, so atoi is enough here.  Just it should check for 0 as error (0 does not make sense anyway here)